src/serde/ser.rs

```diff
@@ -0,0 +1,328 @@
+use core::result::Result;
+
+use serde::{
+    ser::{self, SerializeStruct},
+    Serialize,
+};
+
+use crate::{serde::error::Error, Encode, Value};
+
+pub(crate) struct Serializer {}
```

Comment:
You could define it as:
```suggestion
pub(crate) struct Serializer;
```
To make construction simpler.